src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,321 @@
+import { connectForm, disconnectForm, FormComponent, HiddenFormInputSlot } from "../../utils/form";
+import { connectLabel, disconnectLabel, getLabelText, LabelableComponent } from "../../utils/label";
+import { getSlotted } from "../../utils/dom";
+import { CSS, ERROR_MESSAGES, SLOTS } from "./resources";
+import {
+  connectLocalized,
+  disconnectLocalized,
+  LocalizedComponent,
+  NumberingSystem,
+  numberStringFormatter
+} from "../../utils/locale";
+import { createObserver } from "../../utils/observers";
+@Component({
+  tag: "calcite-textarea",
+  styleUrl: "textarea.scss",
+  shadow: true
+})
+export class Textarea implements FormComponent, LabelableComponent, LocalizedComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteTextareaElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** When `true`, focuses the `textarea` element on page render. */
+  @Prop({ reflect: true }) autofocus: boolean;
+
+  /** When `true`, disables the component. */
+  @Prop({ reflect: true }) disabled: boolean;
+
+  /** Specifies the placeholder text for the input. */
+  @Prop({ reflect: true }) placeholder: string;
+
+  /** Whne `true`, the component's value can be read, but cannot be modified.  */
+  @Prop({ reflect: true }) readonly: boolean;
+
+  /** Specifies number or rows allowed. */
+  @Prop({ reflect: true }) rows: number;
+
+  /** Specifies number or columns allowed. */
+  @Prop({ reflect: true }) cols: number;
+
+  /** Specifies maximum number of characters allowed. */
+  @Prop({ reflect: true }) maxlength: number;
+
+  /** Specifies name of the component  */
+  @Prop({ reflect: true }) name: string;
+
+  /** Specifies the size of `textarea` component. */
+  @Prop({ reflect: true }) scale: "l" | "m" | "s" = "m";
+
+  /** Specifies wrapping mechanism for the text.  */
+  @Prop({ reflect: true }) wrap: "soft" | "hard" = "soft";
+
+  // @Prop() autocomplete: boolean;
+
+  /** The component's value. */
+  @Prop({ mutable: true }) value: string;
+
+  /** When `true` , footer will be added to the component. */
+  @Prop({ reflect: true }) footer = false;
+
+  /** When `true`, disables the resizing handle. */
+  @Prop({ reflect: true }) resizeDisabled: boolean;
+
+  /** When `true`, disables resizing textarea horizantally. */
+  @Prop({ reflect: true }) horizantalResizeDisabled: boolean;
+
+  /** When `true`, disables resizing textarea vertically. */
+  @Prop({ reflect: true }) verticalResizeDisabled: boolean;
+
+  /** When `true`, marks this component as required in form. */
+  @Prop({ reflect: true }) required: boolean;
+
+  /** The id of the form `textarea` is associated with. */
+  @Prop({ reflect: true }) form: string;
```

Comment:
I don't think other Form components have a form id like this. Do we need this?

Reply:
Just checked with other form components. I don't think we need this, FormComponent interface has this built-in.